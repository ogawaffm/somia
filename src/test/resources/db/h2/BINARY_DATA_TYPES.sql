DROP TABLE IF EXISTS BINARY_DATA_TYPES;
CREATE TABLE BINARY_DATA_TYPES (
    -- with default size of 1 byte
    BINARY_COL BINARY,
    BINARY_VARYING_COL BINARY VARYING,
    VARBINARY_COL VARBINARY,

    -- with size of 1 byte
    BINARY_1_COL BINARY(1),
    BINARY_VARYING_1_COL BINARY VARYING(1),
    VARBINARY_1_COL VARBINARY(1),

    -- with maximum size of 1000000000 bytes
    BINARY_1000000000_COL BINARY(1000000000),
    BINARY_VARYING_1000000000_COL BINARY VARYING(1000000000),
    VARBINARY_1000000000_COL VARBINARY(1000000000),

    -- with default size of 1 byte
    BLOB_COL BLOB,
    BINARY_LARGE_OBJECT_COL BINARY LARGE OBJECT,

    -- with default size of 1 byte
    BLOB_1_COL BLOB(1),
    BINARY_LARGE_OBJECT_1_COL BINARY LARGE OBJECT(1),

    -- with maximum size of 1 k bytes
    BLOB_1K_COL BLOB(1 K),
    BINARY_LARGE_OBJECT_1K_COL BINARY LARGE OBJECT(1 K),

    -- with maximum size of 1000000000 bytes
    BLOB_9223372036854775807_COL BLOB(9223372036854775807),
    BINARY_LARGE_OBJECT_9223372036854775807_COL BINARY LARGE OBJECT(9223372036854775807)

)
