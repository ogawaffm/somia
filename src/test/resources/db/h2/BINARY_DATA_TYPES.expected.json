{
  "name" : "BINARY_DATA_TYPES",
  "catalogName" : "TEST",
  "schemaName" : "TEST",
  "typeName" : "BASE TABLE",
  "comment" : "",
  "columns" : [ {
    "name" : "BINARY_COL",
    "type" : {
      "name" : "BINARY",
      "typeNumber" : -2,
      "className" : "byte[]",
      "precision" : 1,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "BINARY_VARYING_COL",
    "type" : {
      "name" : "BINARY VARYING",
      "typeNumber" : -3,
      "className" : "byte[]",
      "precision" : 1000000000,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2000000000,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "VARBINARY_COL",
    "type" : {
      "name" : "BINARY VARYING",
      "typeNumber" : -3,
      "className" : "byte[]",
      "precision" : 1000000000,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2000000000,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "BINARY_1_COL",
    "type" : {
      "name" : "BINARY",
      "typeNumber" : -2,
      "className" : "byte[]",
      "precision" : 1,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "BINARY_VARYING_1_COL",
    "type" : {
      "name" : "BINARY VARYING",
      "typeNumber" : -3,
      "className" : "byte[]",
      "precision" : 1,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "VARBINARY_1_COL",
    "type" : {
      "name" : "BINARY VARYING",
      "typeNumber" : -3,
      "className" : "byte[]",
      "precision" : 1,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "BINARY_1000000000_COL",
    "type" : {
      "name" : "BINARY",
      "typeNumber" : -2,
      "className" : "byte[]",
      "precision" : 1000000000,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2000000000,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "BINARY_VARYING_1000000000_COL",
    "type" : {
      "name" : "BINARY VARYING",
      "typeNumber" : -3,
      "className" : "byte[]",
      "precision" : 1000000000,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2000000000,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "VARBINARY_1000000000_COL",
    "type" : {
      "name" : "BINARY VARYING",
      "typeNumber" : -3,
      "className" : "byte[]",
      "precision" : 1000000000,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2000000000,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "BLOB_COL",
    "type" : {
      "name" : "BINARY LARGE OBJECT",
      "typeNumber" : 2004,
      "className" : "java.sql.Blob",
      "precision" : 2147483647,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : false,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2147483647,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "BINARY_LARGE_OBJECT_COL",
    "type" : {
      "name" : "BINARY LARGE OBJECT",
      "typeNumber" : 2004,
      "className" : "java.sql.Blob",
      "precision" : 2147483647,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : false,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2147483647,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "BLOB_1_COL",
    "type" : {
      "name" : "BINARY LARGE OBJECT",
      "typeNumber" : 2004,
      "className" : "java.sql.Blob",
      "precision" : 1,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : false,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "BINARY_LARGE_OBJECT_1_COL",
    "type" : {
      "name" : "BINARY LARGE OBJECT",
      "typeNumber" : 2004,
      "className" : "java.sql.Blob",
      "precision" : 1,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : false,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "BLOB_1K_COL",
    "type" : {
      "name" : "BINARY LARGE OBJECT",
      "typeNumber" : 2004,
      "className" : "java.sql.Blob",
      "precision" : 1024,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : false,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2048,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "BINARY_LARGE_OBJECT_1K_COL",
    "type" : {
      "name" : "BINARY LARGE OBJECT",
      "typeNumber" : 2004,
      "className" : "java.sql.Blob",
      "precision" : 1024,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : false,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2048,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "BLOB_9223372036854775807_COL",
    "type" : {
      "name" : "BINARY LARGE OBJECT",
      "typeNumber" : 2004,
      "className" : "java.sql.Blob",
      "precision" : 2147483647,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : false,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2147483647,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "BINARY_LARGE_OBJECT_9223372036854775807_COL",
    "type" : {
      "name" : "BINARY LARGE OBJECT",
      "typeNumber" : 2004,
      "className" : "java.sql.Blob",
      "precision" : 2147483647,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : false,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 2147483647,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  } ],
  "indexes" : [ ],
  "collation" : "",
  "primaryKey" : null,
  "bestRowIdentifier" : null
}